replica/table.cc

```diff
@@ -2153,14 +2177,18 @@ future<> compaction_group::stop() noexcept {
     if (_async_gate.is_closed()) {
         co_return;
     }
-    co_await _async_gate.close();
+    auto closed_gate_fut = coroutine::as_future(_async_gate.close());
 
     auto flush_future = co_await seastar::coroutine::as_future(flush());
     co_await _t._compaction_manager.remove(as_table_state());
 
     if (flush_future.failed()) {
         co_await seastar::coroutine::return_exception_ptr(flush_future.get_exception());
     }
+    auto closed_gate_fut_res = co_await std::move(closed_gate_fut);
+    if (closed_gate_fut_res.failed()) {
```

Comment:
keeping the as_future_awaiter in a variable is bit weird. my bad for suggesting it.
I take my words back. If we want `_async_gate.close()` to happen in the background, it's ok to do:
```
    auto closed_gate_fut = _async_gate.close();
...
    co_await std::move(closed_gate_fut);
```

we're in stop so stop() is never expected to fail, and if it did, it's ok to throw inside the coroutine.
we're not on the hot path anyways.

so let's simplify this.
again, I apologize for going back and forth on this